packages/react-components/src/organisms/__tests__/Form.test.tsx

```diff
@@ -0,0 +1,274 @@
+      const handleSave = jest.fn(() => Promise.resolve());
+      const handleValidate = jest.fn(() => false);
+      const { getByText } = render(
+        <Form {...props} validate={handleValidate} onSave={handleSave} dirty>
+          {({ onSave: onSubmit }) => (
+            <>
+              <input type="text" />
+              <Button primary onClick={onSubmit}>
+                save
+              </Button>
+            </>
+          )}
+        </Form>,
+        { wrapper: StaticRouter },
+      );
+
+      userEvent.click(getByText(/^save/i));
+
+      expect(handleValidate).toHaveBeenCalled();
+      expect(handleSave).not.toHaveBeenCalled();
+    });
+  });
+
+  describe('and the form is valid', () => {
+    let result!: RenderResult;
+
+    let handleSave!: jest.MockedFunction<() => Promise<void>>;
+    let resolveSave!: () => void;
+    let rejectSave!: (error: Error) => void;
+
+    beforeEach(() => {
+      handleSave = jest.fn().mockReturnValue(
+        new Promise<void>((resolve, reject) => {
+          resolveSave = resolve;
+          rejectSave = reject;
+        }),
+      );
+      history = createMemoryHistory({ getUserConfirmation });
+      result = render(
+        <Router history={history}>
+          <Form {...props} onSave={handleSave} dirty>
+            {({ onSave: onSubmit, isSaving }) => (
+              <>
+                <Link to={'/another-url'}>Navigate away</Link>
+                <Button primary enabled={!isSaving} onClick={onSubmit}>
+                  save
+                </Button>
+              </>
+            )}
+          </Form>
+        </Router>,
+      );
+    });
+
+    it('calls onSave', async () => {
+      const { getByText } = result;
+
+      userEvent.click(getByText(/^save/i));
+      expect(handleSave).toHaveBeenCalled();
+
+      act(resolveSave);
+      await waitFor(() =>
+        expect(getByText(/^save/i).closest('button')).toBeEnabled(),
+      );
+    });
+
+    it('disables the save button while saving', async () => {
+      const { getByText, unmount } = result;
+
+      userEvent.click(getByText(/^save/i));
+      expect(getByText(/^save/i).closest('button')).toBeDisabled();
+
+      unmount();
+      resolveSave();
+    });
+
+    it('prompts when trying to leave while saving', async () => {
+      const { getByText, unmount } = result;
+      userEvent.click(getByText(/^save/i));
+
+      userEvent.click(getByText(/navigate/i));
+      expect(getUserConfirmation).toHaveBeenCalled();
+
+      unmount();
+      resolveSave();
+    });
+
+    describe('and the save succeeds', () => {
+      beforeEach(() => {
+        act(resolveSave);
+      });
+
+      it('resets to initial state once a saved form matches the default state', async () => {
+        const { getByText, rerender } = result;
+
+        userEvent.click(getByText(/^save/i));
+        await waitFor(() =>
+          expect(getByText(/^save/i).closest('button')).toBeEnabled(),
+        );
+        rerender(
```

Comment:
Does this test anything? Isn't this the same as 're-enables the save button'?

Reply:
This is unfortunately for coverage right now. I will refactor this out once I modify EditModal to reuse this component. I just want to be able to merge to not block new form work